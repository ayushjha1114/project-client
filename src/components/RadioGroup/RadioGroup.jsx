import React from 'react';
import PropTypes from 'prop-types';
import style from './style';

const propTypes = {
  error: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.object),
};

// default values for props:
const defaultProps = {
  error: '',
  options: [],
  value: '',
  onChange: () => {},
};

class RadioGroup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      error, options, ...rest
    } = this.props;
    const css = `
    .radio{
        cursor: pointer;
    }`;
    return (
      <>
        <style>{css}</style>
        <div>
          <h4> What do you do?</h4>
          {options.map(item => (
            <div key={item.label}>
              <label htmlFor="asf">
                <input type="radio" name="sport" {...rest} key={item.label} value={item.label} />
                <span className="radio">{item.label}</span>
              </label>
            </div>
          ))}
          { (error) ? <p style={{ ...style.para }}>{error}</p> : '' }
        </div>
      </>
    );
  }
}

RadioGroup.propTypes = propTypes;
RadioGroup.defaultProps = defaultProps;

export default RadioGroup;
