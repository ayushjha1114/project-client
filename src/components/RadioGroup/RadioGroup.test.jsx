import React from 'react';
import { shallow, mount } from 'enzyme';
import RadioGroup from './RadioGroup';

describe('RadioGroup', () => {
  test('should render correctly', () => {
    const component = shallow(<RadioGroup />);
    expect(component).toMatchSnapshot();
  });
  it('check options props', () => {
    const props = {
      options: [],
    };

    const component = mount(<RadioGroup {...props} />);
    expect(component.prop('options')).toEqual([]);
  });
});
