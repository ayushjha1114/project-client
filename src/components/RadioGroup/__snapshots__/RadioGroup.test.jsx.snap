// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroup
    error=""
    onChange={[Function]}
    options={Array []}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <style>
          
    .radio{
        cursor: pointer;
    }
        </style>,
        <div>
          <h4>
             What do you do?
          </h4>
          
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "
    .radio{
        cursor: pointer;
    }",
        },
        "ref": null,
        "rendered": "
    .radio{
        cursor: pointer;
    }",
        "type": "style",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
               What do you do?
            </h4>,
            Array [],
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " What do you do?",
            },
            "ref": null,
            "rendered": " What do you do?",
            "type": "h4",
          },
          "",
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <style>
            
    .radio{
        cursor: pointer;
    }
          </style>,
          <div>
            <h4>
               What do you do?
            </h4>
            
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
    .radio{
        cursor: pointer;
    }",
          },
          "ref": null,
          "rendered": "
    .radio{
        cursor: pointer;
    }",
          "type": "style",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                 What do you do?
              </h4>,
              Array [],
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " What do you do?",
              },
              "ref": null,
              "rendered": " What do you do?",
              "type": "h4",
            },
            "",
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
